pkg/client/record/event.go

```diff
@@ -266,13 +276,22 @@ func (recorder *recorderImpl) generateEvent(object runtime.Object, timestamp unv
 	event := recorder.makeEvent(ref, eventtype, reason, message)
 	event.Source = recorder.source
 
+	recorder.wg.Add(2)
```

Comment:
> adding 2 because we want to wait for the event to be recorded to broadcaster but also processed using recordToSink

I don't think this does what you want.  I'm can call `StartRecordingToSink` multiple times.  Doing that will mean that you have to wait for all of them to record, not just one.